### ISSUE TITLE
JobClient.waitForCompletion throws an exception if getJobStatus times out

### ISSUE THREAD

**Issue by danielbwatson** [Netflix/genie#481]
waitForCompletion calls getJobStatus but it doesn't not wrap the call in a check for exceptions.  The current implementation can throw a SocketTimeout exception when getting the status from the Genie server.

```text
Caused by: java.net.SocketTimeoutException: timeout
	at okio.Okio$3.newTimeoutException(Okio.java:212)
	at okio.AsyncTimeout.exit(AsyncTimeout.java:288)
	at okio.AsyncTimeout$2.read(AsyncTimeout.java:242)
	at okio.RealBufferedSource.indexOf(RealBufferedSource.java:325)
	at okio.RealBufferedSource.indexOf(RealBufferedSource.java:314)
	at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:210)
	at okhttp3.internal.http.Http1xStream.readResponse(Http1xStream.java:184)
	at okhttp3.internal.http.Http1xStream.readResponseHeaders(Http1xStream.java:125)
	at okhttp3.internal.http.HttpEngine.readNetworkResponse(HttpEngine.java:775)
	at okhttp3.internal.http.HttpEngine.access$200(HttpEngine.java:86)
	at okhttp3.internal.http.HttpEngine$NetworkInterceptorChain.proceed(HttpEngine.java:760)
	at okhttp3.internal.http.HttpEngine.readResponse(HttpEngine.java:613)
	at okhttp3.RealCall.getResponse(RealCall.java:244)
	at okhttp3.RealCall$ApplicationInterceptorChain.proceed(RealCall.java:201)
	at com.netflix.genie.client.interceptors.UserAgentInsertInterceptor.intercept(UserAgentInsertInterceptor.java:68)
	at okhttp3.RealCall$ApplicationInterceptorChain.proceed(RealCall.java:190)
	at com.netflix.genie.client.security.oauth2.impl.OAuth2SecurityInterceptor.intercept(OAuth2SecurityInterceptor.java:77)
	at okhttp3.RealCall$ApplicationInterceptorChain.proceed(RealCall.java:190)
	at com.netflix.genie.client.interceptors.ResponseMappingInterceptor.intercept(ResponseMappingInterceptor.java:51)
	at okhttp3.RealCall$ApplicationInterceptorChain.proceed(RealCall.java:190)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:163)
	at okhttp3.RealCall.execute(RealCall.java:57)
	at retrofit2.OkHttpCall.execute(OkHttpCall.java:174)
	at com.netflix.genie.client.JobClient.getJobStatus(JobClient.java:373)
	at com.netflix.genie.client.JobClient.waitForCompletion(JobClient.java:419)
	at com.netflix.genie.client.JobClient$waitForCompletion$0.call(Unknown Source)
```

**Comment by tgianos** [comment]
@danielbwatson this fix is now part of the 3.0.3 release. If you believe that solves your issue please close this issue. Thanks.

**Comment by danielbwatson** [comment]
Fixed in release 3.0.3.
